Technical review: How it Works (#6)
askui/docs-mintlify: introduction/01-introduction/03-how-it-works.mdx

AskUI consists of mutliple components which all work together to allow users to build versatile automations across different operating systems. These components are:

> * **Model Provider:** The Model Providers host AI models. The AI models are the brain of the Agents for Locating Elements (Grounding Models), Asking (Query Models) and autonoumous Acting (Large Action Models)

Reviewer: ```suggestion
* **Model Provider:** The model providers host AI models. These models are the brain of the agents for locating elements (Grounding Models), asking (Query Models) and autonomous acting (Large Action Models)
```

> * **Agent Framework:** The Agent Framework is used to build an Vision Agent in Python. You can easily interact with any OS by taking over the mouse and keyborad, visulizing the thinking process or use specialized tools like excel, pdf or web requests. 

Reviewer: ```suggestion
* **Agent Framework:** The agent framework is used to build an Vision Agent in Python. You can easily interact with any OS by taking over the mouse and keyboard, visualizing the thinking process or use specialized tools like excel, pdf or web requests.
```

> * **Agent OS:** The Agent OS is installed on the device to automated, executing all commands and interacting with mouse and keyboard and visulize the thinking process. Serves as the “hands” and "speech" of your Vision Agent.

Reviewer: ```suggestion
* **Agent OS:** The Agent OS is installed on the device to be automated, executing all commands and interacting with mouse and keyboard and visualize the thinking process.
```

> * **Agent Hub Web:** The Agent Hub Web provides you an easy way to coloborate with you're team. It allows you to sync your build Vision Agents over the repository to other devices. Let's trigger you agents from E-Mail or Web, schedule Agents on remote devices and manage your team.

Reviewer: ```suggestion
* **Agent Hub Web:** The Agent Hub Web provides you an easy way to collaborate with your team. It allows you to synchronize your Vision Agents via the repository to other devices. Trigger you agents from email or web, schedule agents on remote devices and manage your team.
```

> * **Agent Hub Desktop:** The Agent Hub Desktop is GUI for your Vision Agents. This is installed on your local device and allows you to setup your device, sync agents and interact with running agents. To be released Summer 2025.&#x20;

Reviewer: ```suggestion
* **Agent Hub Desktop:** The Agent Hub Desktop is a graphical interface for your Vision Agents. This is installed on your local device and allows you to setup your device, sync agents and interact with running agents. To be released Summer 2025.&#x20;
```

Lets have a deeper look at these components:

## Building Your Agent

The **Agent Framework** is where you **tell AskUI what to automate**. You can write single-step instructions or let the agent act on its own e.g. `agent.act("Book me flight from Karlsruhe to Rome")`.

For example, let’s say you want AskUI to click a "Login" button (single-step action). You can define your agent in python e.g. `my_agent.py`:
```python
with VisionAgent(display=1) as agent: 
    agent.click("the Login button")
```
* **`VisionAgent(display=1)`** → Start and configure the agent (e.g. start a Vision Agent on display 1).
* **`agent.click("Element locator")`** → Specifies the action of the agent (clicking)

## Running your Agent

> Here’s what’s happening under the hood when you now execute the agent with `python my_agent.py`:

Reviewer: ```suggestion
Here’s what’s happening under the hood when you execute the agent with python my_agent.py now:
```
